src/algebraic_topology/dold_kan/homotopies.lean

```diff
@@ -0,0 +1,204 @@
+In the case `C` is abelian, this `P_infty` shall be the projection on the
+normalized Moore subcomplex of `K[X]` associated to the decomposition of the
+complex `K[X]` as a direct sum of this normalized subcomplex and of the
+degenerate subcomplex.
+
+In `p_infty.lean`, this endomorphism `P_infty` shall be obtained by
+passing to the limit idempotent endomorphisms `P q` for all `(q : ℕ)`.
+These endomorphisms `P q` are defined by induction. The idea is to
+start from the identity endomorphism `P 0` of `K[X]` and to ensure by
+induction that the `q` higher face maps (except $d_0$) vanish on the
+image of `P q`. Then, in a certain degree `n`, the image of `P q` for
+a big enough `q` will be contained in the normalized subcomplex. This
+construction is done in `projections.lean`.
+
+It would be easy to define the `P q` degreewise (similarly as it is done
+in *Simplicial Homotopy Theory* by Goerrs-Jardine p. 149), but then we would
+have to prove that they are compatible with the differential (i.e. they
+are chain complex maps), and also that they are homotopic to the identity.
+These two verifications are quite technical. In order to reduce the number
+of such technical lemmas, the strategy that is followed here is to define
+a series of null homotopic maps `Hσ q` (attached to families of maps `hσ`)
+and use these in order to construct `P q` : the endomorphisms `P q`
+shall basically be obtained by altering the identity endomorphism by adding
+null homotopic maps, so that we get for free that they are morphisms
+of chain complexes and that they are homotopic to the identity. The most
+technical verifications that are needed about the null homotopic maps `Hσ`
+are obtained in `faces.lean`.
+
+In this file `homotopies.lean`, we define the null homotopic maps
+`Hσ q : K[X] ⟶ K[X]`, show that they are natural (see `nat_trans_Hσ`) and
+compatible the application of additive functors (see `map_Hσ`).
+
+## References
+* Albrecht Dold, Homology of Symmetric Products and Other Functors of Complexes,
+Annals of Mathematics, Second Series, Vol. 68 No. 1 (Jul. 1958), pp. 54-80.
+* Paul G. Goerss, John F. Jardine, Simplical Homotopy Theory, Modern Birkhäuser Classics,
+Reprint of the 1999 edition.
+
+-/
+
+open category_theory
+open category_theory.category
+open category_theory.limits
+open category_theory.preadditive
+open category_theory.simplicial_object
+open homotopy
+open opposite
+open_locale simplicial dold_kan
+
+noncomputable theory
+
+namespace algebraic_topology
+
+namespace dold_kan
+
+variables {C : Type*} [category C] [preadditive C]
+variables {X : simplicial_object C}
+
+/-- As we are using chain complexes indexed by `ℕ`, we shall need the relation
+`c` such `c m n` if and only if `n+1=m`. -/
+abbreviation c := complex_shape.down ℕ
+
+/-- Helper when we need some `c.rel i j` (i.e. `complex_shape.down ℕ`),
+e.g. `c_mk n (n+1) rfl` -/
+lemma c_mk (i j : ℕ) (h : j+1 = i) : c.rel i j := h
```

Comment:
As this is generally useful, perhaps we could provide it as a lemma attached to `complex_shape.down`, and then just use an abbreviation (or, even better, a notation?) to point to it here.

Reply:
Ok. I have changed it into an abbreviation.